src/misc/cross.cpp

```diff
@@ -439,6 +440,224 @@ FILE *fopen_wrap_ro_fallback(const std::string &filename, bool &is_readonly)
+		r++;
+	}
+	r = 0;
+	while (r < DOS_EXTLENGTH) {
+		if (wild_ext[r] == '*')
+			return true;
+		if (wild_ext[r] != '?' && wild_ext[r] != file_ext[r])
+			return false;
+		r++;
+	}
+	return true;
+}
+
+bool LWildFileCmp(const char *file, const char *wild)
+{
+	if (*file == 0)
+		return false;
+	char file_name[LFN_NAMELENGTH + 1];
+	char file_ext[LFN_NAMELENGTH + 1];
+	char wild_name[LFN_NAMELENGTH + 1];
+	char wild_ext[LFN_NAMELENGTH + 1];
+	Bitu r;
+
+	for (r = 0; r <= LFN_NAMELENGTH; r++) {
+		file_name[r] = 0;
+		wild_name[r] = 0;
+		file_ext[r] = 0;
+		wild_ext[r] = 0;
+	}
+
+	Bitu size;
+	size_t elength;
+	const char *find_ext;
+	find_ext = strrchr(file, '.');
+	if (find_ext) {
+		size = (Bitu)(find_ext - file);
+		if (size > LFN_NAMELENGTH)
+			size = LFN_NAMELENGTH;
+		memcpy(file_name, file, size);
+		find_ext++;
+		elength = strlen(find_ext);
+		memcpy(file_ext, find_ext,
+		       (strlen(find_ext) > LFN_NAMELENGTH) ? LFN_NAMELENGTH
+		                                           : strlen(find_ext));
+	} else {
+		size = strlen(file);
+		elength = 0;
+		memcpy(file_name, file,
+		       (strlen(file) > LFN_NAMELENGTH) ? LFN_NAMELENGTH
+		                                       : strlen(file));
+	}
+	upcase(file_name);
+	upcase(file_ext);
+	char nwild[LFN_NAMELENGTH + 2];
+	strcpy(nwild, wild);
+	if (strrchr(nwild, '*') && strrchr(nwild, '.') == NULL)
+		strcat(nwild, ".*");
+	find_ext = strrchr(nwild, '.');
+	if (find_ext) {
+		if (wild_match(file, nwild))
+			return true;
+		Bitu size = (Bitu)(find_ext - nwild);
+		if (size > LFN_NAMELENGTH)
+			size = LFN_NAMELENGTH;
+		memcpy(wild_name, nwild, size);
+		find_ext++;
+		memcpy(wild_ext, find_ext,
+		       (strlen(find_ext) > LFN_NAMELENGTH) ? LFN_NAMELENGTH
+		                                           : strlen(find_ext));
+	} else {
+		memcpy(wild_name, nwild,
+		       (strlen(nwild) > LFN_NAMELENGTH) ? LFN_NAMELENGTH
+		                                        : strlen(nwild));
+	}
+	upcase(wild_name);
+	upcase(wild_ext);
+	/* Names are right do some checking */
+	if (strchr(wild_name, '*')) {
+		if (!strchr(wild, '.'))
+			return wild_match(file, wild_name);
+		else if (!wild_match(file_name, wild_name))
+			return false;
+	} else {
+		r = 0;
+		while (r < size) {
+			if (wild_name[r] == '*')
+				break;
+			if (wild_name[r] != '?' && wild_name[r] != file_name[r])
+				return false;
+			r++;
+		}
+		if (wild_name[r] && wild_name[r] != '*')
+			return false;
+	}
+	if (strchr(wild_ext, '*'))
+		return wild_match(file_ext, wild_ext);
+	else {
+		r = 0;
+		while (r < elength) {
+			if (wild_ext[r] == '*')
+				return true;
+			if (wild_ext[r] != '?' && wild_ext[r] != file_ext[r])
+				return false;
+			r++;
+		}
+		if (wild_ext[r] && wild_ext[r] != '*')
+			return false;
+		return true;
+	}
+}
+
+bool get_expanded_files(const std::string &path,
+                        std::vector<std::string> &files,
+                        bool files_only) noexcept
+{
+	files.clear();
```

Comment:
If the existing `paths` vector is passed in here (instead of files), then we can create the temporary vector `files` here instead, instead of clearing it.  And then only at the bottom of this function, you can append to the `paths` vector when the conditions are correct (ie: files.size() > 0).

Reply:
Done.